pkg/doc/verifiable/credential.go

```diff
@@ -276,20 +328,33 @@ func decodeIssuer(data []byte, credential *Credential) error {
 
 // NewCredential creates an instance of Verifiable Credential by reading a JSON document from bytes.
 // It also applies miscellaneous options like custom decoders or settings of schema validation.
-func NewCredential(dataJSON []byte, opts ...CredentialOpt) (*Credential, error) {
+func NewCredential(vcData []byte, opts ...CredentialOpt) (*Credential, error) {
 	// Apply options
 	crOpts := defaultCredentialOpts()
 	for _, opt := range opts {
 		opt(crOpts)
 	}
 
-	raw := &rawCredential{}
-	err := json.Unmarshal(dataJSON, &raw)
-	if err != nil {
-		return nil, fmt.Errorf("JSON unmarshalling of verifiable credential failed: %w", err)
+	var raw *rawCredential
+	var err error
+	if crOpts.issuerPublicKeyFetcher != nil {
+		var vcDataFromJwt []byte
+		if vcDataFromJwt, raw, err = decodeJws(vcData, crOpts.verifyJwtSignature, crOpts.issuerPublicKeyFetcher); err != nil {
+			logger.Infof("cannot decode JWS: %w", err)
+		} else {
+			vcData = vcDataFromJwt
```

Comment:
As a best practice, it's best not to reassign input parameters.

Reply:
Got it! I will change it, thanks!